heron/spi/src/java/com/twitter/heron/spi/utils/NetworkUtils.java

```diff
@@ -229,9 +250,25 @@ public static boolean sendHttpGetRequest(HttpURLConnection connection) {
     }
   }
 
-  public static HttpURLConnection getConnection(String endpoint) throws IOException {
-    URL url = new URL(endpoint);
-    return (HttpURLConnection) url.openConnection();
+  public static HttpURLConnection getHttpConnection(String endpoint) {
+    // construct the http url connection
+    try {
+      URL url = new URL(endpoint);
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
We need some defensive coding here before assuming that the connection is a HTTPURLConnection